sn_networking/src/event.rs

```diff
@@ -502,24 +502,43 @@ impl SwarmDriver {
             } => {
                 event_string = "OutgoingConnErr";
                 warn!("OutgoingConnectionError to {failed_peer_id:?} on {connection_id:?} - {error:?}");
-                // info!("{:?}", self.swarm.network_info());
 
                 // we need to decide if this was a critical error and the peer should be removed from the routing table
                 let should_clean_peer = match error {
                     DialError::Transport(errors) => {
                         // as it's an outgoing error, if it's transport based we can assume it is _our_ fault
+                        //
                         // (eg, could not get a port for a tcp connection)
                         // so we default to it not being a real issue
-                        let mut there_is_a_real_issue = false;
+                        // unless there are _specific_ errors (connection refused eg)
+                        error!("Dial errors len : {:?}", errors.len());
+                        let mut there_is_a_serious_issue = false;
                         for (_addr, err) in errors {
                             error!("OutgoingTransport error : {err:?}");
-                            if let TransportError::MultiaddrNotSupported(addr) = err {
-                                warn!("Multiaddr not supported : {addr:?}");
-                                // if we can't dial a peer on a given address, we should remove it from the routing table
-                                there_is_a_real_issue = true
+
+                            match err {
+                                TransportError::MultiaddrNotSupported(addr) => {
+                                    warn!("Multiaddr not supported : {addr:?}");
+                                    // if we can't dial a peer on a given address, we should remove it from the routing table
+                                    there_is_a_serious_issue = true
+                                }
+                                TransportError::Other(err) => {
+                                    let problematic_errors =
+                                        ["ConnectionReset", "ConnectionRefused", "HostUnreachable"];
```

Comment:
Not sure about the `ConnectionReset` .
If say the peer closed a connection due to long time of connection, say we have `disconnect outdated connection for non-RT nodes` , will it be counted as `ConnectionReset` ?